worker/src/queues/legacyIngestionQueue.ts

```diff
@@ -8,21 +8,77 @@ import {
   recordHistogram,
   TQueueJobTypes,
   logger,
+  IngestionEventType,
+  S3StorageService,
+  ingestionBatchEvent,
+  ingestionEvent,
 } from "@langfuse/shared/src/server";
 
 import {
   handleBatch,
-  sendToWorkerIfEnvironmentConfigured,
+  addTracesToTraceUpsertQueue,
 } from "@langfuse/shared/src/server";
 import { tokenCount } from "../features/tokenisation/usage";
+import { env } from "../env";
 
 export const legacyIngestionQueueProcessor: Processor = async (
   job: Job<TQueueJobTypes[QueueName.LegacyIngestionQueue]>,
 ) => {
   try {
     const startTime = Date.now();
+
+    let ingestionEvents: IngestionEventType[] = [];
+    if (job.data.payload.useS3EventStore) {
+      if (
+        env.LANGFUSE_S3_EVENT_UPLOAD_ENABLED !== "true" ||
+        !env.LANGFUSE_S3_EVENT_UPLOAD_BUCKET
+      ) {
+        throw new Error(
+          "S3 event store is not enabled but useS3EventStore is true",
+        );
+      }
+      // If we used the S3 store we need to fetch the ingestionEvents from S3
+      const s3Client = new S3StorageService({
```

Comment:
How efficient is it to always re-instantiate the service and also within the S3 client? Should this be treated as singleton to make use of longer lasting TCP connections?

Reply:
Addressed in https://github.com/langfuse/langfuse/pull/3537/commits/61b5623302e2fb47841110820913ae2b04774065. Good catch!